docs/core/magnetic_field.md

```diff
@@ -0,0 +1,228 @@
+
+:::{doxygenfunction} Acts::MagneticFieldProvider::getField
+:outline:
+:::
+
+Aside from the lookup position
+as a global vector, it accepts an instance of the opaque cache object mentioned
+before. The return value is a {class}`Acts::Result` object. It either contains
+the field value at the requested location, or an
+{enum}`Acts::MagneticFieldError` in case of a lookup failure, like an
+out-of-bounds lookup position.
+
+Below is an example of how a client can interact with an instance of {class}`Acts::MagneticFieldProvider`.
+
+```cpp
+// in event context
+auto fieldContext = getExperimentFieldContext();
+const Acts::MagneticFieldProvider& fieldProvider = getFieldProvider();
+auto cache = fieldProvider.makeCache(fieldContext);
+
+auto lookupResult = fieldProvider.getField(Acts::Vector3{10, 10, 10}, cache);
+if(!lookupResult.ok()) {
+   throw std::runtime_error{"Field lookup failure"};
+}
+
+Acts::Vector3 fieldValue = *lookupResult;
+```
+
+## Magnetic field context
+
+:::{doxygentypedef} Acts::MagneticFieldContext
+:outline:
+:::
+
+The magnetic field context is an opaque type which contains experiment specific
+event context information. This can be used to supply event dependent data to
+the magnetic field instance, in case it is needed to provide correct field
+values. The library itself does not make any assumptions on the content of this
+context type (it is implemented using `std::any`), but passes a reference
+through the call-chain to the field implementation. An experiment specific
+field implementation is then expected to performa cast to the concrete type,
+and use the contents. 
+
+An example of this is a conditions database handle that is derived from the
+event context, and is used by the magnetic field instance to retrieve the
+correct field values.
+
+## Field provider implementations in Core
+
+There are a number of field provider implemenations found in core which serve different purposes.
+
+### Constant magnetic field
+
+The simplest implementation is a constant field, which returns the same field values at every queried location. It is implemented in the {class}`Acts::ConstantBField` class.
+
+:::{doxygenclass} Acts::ConstantBField
+:members: ConstantBField
+:::
+
+As seen above, the class is constructed from a three-dimensional field vector, which is returned unmodified to every call to {func}`Acts::ConstantBField::getField`.
+
+### Interpolated magnetic field
+
+For more complex magnetic field implementations
+{class}`Acts::InterpolatedMagneticField` can be used. The idea here is to calculate
+an interpolated value of the magnetic field from a grid of known field values.
+In 3D, this means the interpolation is done from the 8 cornerpoints of a *field
+cell*. The field cell can be retrieved for any given position. Since during
+typical access patterns, e.g. the propagation, subsequent steps are relatively
+likely to not cross the field cell boundary, the field cell can be cached.
```

Comment:
be cached. -> be cached and thus cell lookup avoided which improves performance.

Reply:
ah, maybe not, it is said just few lines later.